modules/apps/portal-search/portal-search-test/src/testIntegration/java/com/liferay/portal/search/facet/faceted/searcher/test/CategoryFacetTest.java

```diff
@@ -101,78 +105,137 @@ public void testAvoidResidualDataFromDDMStructureLocalServiceTest()
+		_addCategory("To Do");
 
-		AssetCategory assetCategory = addCategory(title);
+		_addUser();
 
-		long categoryId = assetCategory.getCategoryId();
+		SearchContext searchContext = _getSearchContext();
 
-		addUser(_group, categoryId);
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
 
-		SearchContext searchContext = getSearchContext(
-			assetCategory.getTitleCurrentValue());
+		searchContext.addFacet(facet);
 
-		searchContext.setCategoryIds(new long[] {categoryId});
-		searchContext.setGroupIds(new long[] {_group.getGroupId()});
+		FacetsAssert.assertFrequencies(
+			facet.getFieldName(), searchContext, search(searchContext),
+			Collections.singletonMap(_getAssetVocabularyCategoryId(), 1));
+	}
 
-		Facet facet = categoryFacetFactory.newInstance(searchContext);
+	@Test
+	public void testSelection() throws Exception {
+		_addCategory(RandomTestUtil.randomString());
+
+		_addJournalArticle();
+
+		_addUser();
+
+		SearchContext searchContext = _getSearchContext();
+
+		Facet facet = _categoryFacetFactory.newInstance(searchContext);
+
+		facet.select(_getAssetVocabularyCategoryId());
 
 		searchContext.addFacet(facet);
 
 		Hits hits = search(searchContext);
 
-		Map<String, Integer> frequencies = Collections.singletonMap(
-			_getAssetVocabularyCategoryId(assetCategory), 1);
+		_assertEntryClassNames(
+			Collections.singletonList(User.class.getName()), hits,
+			searchContext);
 
 		FacetsAssert.assertFrequencies(
-			facet.getFieldName(), searchContext, hits, frequencies);
+			facet.getFieldName(), searchContext, hits,
+			Collections.singletonMap(_getAssetVocabularyCategoryId(), 1));
 	}
 
-	protected AssetCategory addCategory(String title) throws Exception {
-		AssetCategory assetCategory = assetCategoryLocalService.addCategory(
+	private void _addCategory(String title) throws Exception {
+		_assetCategory = _assetCategoryLocalService.addCategory(
 			TestPropsValues.getUserId(), _group.getGroupId(), title,
 			_assetVocabulary.getVocabularyId(),
 			ServiceContextTestUtil.getServiceContext(
 				_group.getGroupId(), TestPropsValues.getUserId()));
+	}
 
-		_assetCategory = assetCategory;
-
-		return assetCategory;
+	private void _addJournalArticle() throws Exception {
+		journalArticleSearchFixture.addArticle(
+			new JournalArticleBlueprint() {
+				{
+					setGroupId(_group.getGroupId());
+					setJournalArticleContent(
+						new JournalArticleContent() {
+							{
+								put(
+									LocaleUtil.US,
+									RandomTestUtil.randomString());
+
+								setDefaultLocale(LocaleUtil.US);
+								setName("content");
+							}
+						});
+					setJournalArticleTitle(
+						new JournalArticleTitle() {
+							{
+								put(
+									LocaleUtil.US,
+									_assetCategory.getTitleCurrentValue());
+							}
+						});
+				}
+			});
 	}
 
-	protected void addUser(Group group, long... categoryIds) throws Exception {
-		_user = UserTestUtil.addUser(group.getGroupId());
+	private void _addUser() throws Exception {
```

Comment:
All tests in this class are using this function to add a user. You could reduce the complexity of the code by adding the user in the setUp function. See https://github.com/AlmirFe/liferay-portal/commit/b82762fc9de2f4cae10c353924a26a4f12b1b2d1